code/modules/clothing/modular/modular.dm

```diff
@@ -0,0 +1,217 @@
+	siemens_coefficient = 0.9
+	permeability_coefficient = 1
+	gas_transfer_coefficient = 1
+
+	actions_types = list(/datum/action/item_action/toggle)
+
+	/// An assoc list of stat mods
+	var/list/stat_mod
+
+	/// The human that is wearing the armor
+	var/mob/living/carbon/human/owner
+
+	/// Attachment slots for chest armor
+	var/obj/item/armor_module/armor/slot_chest
+	/// Attachment slots for arm armor
+	var/obj/item/armor_module/armor/slot_arms
+	/// Attachment slots for leg armor
+	var/obj/item/armor_module/armor/slot_legs
+
+	/** Installed modules */
+	/// How many modules you can have
+	var/max_modules = 1
+	/// What modules are installed
+	var/list/obj/item/armor_module/attachable/installed_modules
+	/// What storage is installed
+	var/obj/item/armor_module/storage/installed_storage
+	/// Holder for the actual storage implementation
+	var/obj/item/storage/internal/storage
+
+	///List of allowed types [/obj/item/armor_module] that can be installed into this modular armor.
+	var/allowed_armor_path = list(
+		/obj/item/armor_module,
+	)
+ 	///List of allowed types [/obj/item/armor_module] that can be installed into this modular armor.
+	var/allowed_modules_path = list(
+		/obj/item/armor_module,
+	)
+	///List of allowed types [/obj/item/armor_module] that can be installed into this modular armor.
+	var/allowed_storage_path = list(
+		/obj/item/armor_module,
+	)
+
+	/// Misc stats
+	light_strength = 4
+
+
+/obj/item/clothing/suit/modular/Initialize()
+	. = ..()
+	installed_modules = list()
+
+
+/obj/item/clothing/suit/modular/Destroy()
+	QDEL_NULL(slot_chest)
+	QDEL_NULL(slot_arms)
+	QDEL_NULL(slot_legs)
+
+	QDEL_NULL(installed_modules)
+	QDEL_NULL(installed_storage)
+	QDEL_NULL(storage)
+	return ..()
+
+
+/obj/item/clothing/suit/modular/attack_self(mob/user)
+	if(!isturf(user.loc))
+		to_chat(user, "<span class='warning'>You cannot turn the light on while in [user.loc].</span>")
+		return
+	if(cooldowns[COOLDOWN_ARMOR_LIGHT] || !ishuman(user))
+		return
+	var/mob/living/carbon/human/H = user
+	if(H.wear_suit != src)
+		return
+	toggle_armor_light(user)
+	return TRUE
+
+/obj/item/clothing/suit/modular/item_action_slot_check(mob/user, slot)
+	if(!light_strength) // No light no ability
+		return FALSE
+	if(!ishuman(user))
+		return FALSE
+	if(slot != SLOT_WEAR_SUIT)
+		return FALSE
+	return TRUE //only give action button when armor is worn.
+
+
+/obj/item/clothing/suit/modular/attack_hand(mob/living/user)
+	if(!storage)
+		return ..()
+	if(storage.handle_attack_hand(user))
+		return ..()
+
+
+/obj/item/clothing/suit/modular/MouseDrop(over_object, src_location, over_location)
+	if(!storage)
+		return ..()
+	if(storage?.handle_mousedrop(usr, over_object))
+		return ..()
+
+
+/obj/item/clothing/suit/modular/attackby(obj/item/I, mob/user, params)
+	. = ..()
+	if(.)
+		return
+	if(!storage)
+		return
+	if(I.tool_behaviour)
+		return
```

Comment:
This should already be handled before `attackby` happens, though the melee attack chain thingy.

Reply:
Without this with storage you would try to put the toolin the storage of the armor instead of removing things.

Reply:
That's because `crowbar_act` is not returning `TRUE`

Reply:
oof my bad